Enable API Gateway logging in CloudWatch (#3054) (#3415)
DataBiosphere/azul: README.md

the bucket, you may want to include the region name at then end of the bucket
name. That way you can have consistent bucket names across regions.

Create an IAM role to be used by API Gateway in order to enable CloudWatch logs.
This needs to be done once per region in which CloudWatch logs are enabled.

1. Navigate to the *IAM Management Console*

2. Click *Roles* under *Access management*

3. Click *Create role*

4. Click *AWS service* for *Select type of trusted entity*

5. Click *API Gateway* for *Choose a use case*, click *Next: Permissions*

6. Ensure *AmazonAPIGatewayPushToCloudWatchLogs* is listed exclusively as the
policy and click *Next: Tags*

7. Add *name: azul-api_gateway* as a tag and click *Next: Review*

8. For *Role name* enter `azul-api_gateway`, ensure *Policies* are as prescribed
in step 6 and click *Create role*

9. Be sure to copy the *Role ARN* for the newly created `azul-api_gateway` role
and go to the *API Gateway* console

10. Select any available Gateway resource e.g. `azul-{lambda}-{stage}`, click on
*Settings* at the bottom of the left menu, paste the copied *Role ARN* into the

> *CloudWatch log role ARN* and click *Save*

Reviewer: I'm confused about this step. You're supposed to set the ARN for *any* available Gateway resource? You do it for one and it activates the logging role for everything? Can you explain how this works?

Author: note to self: Add `to modify region wide settings do…` or something like it.

for *Number of days after objects become previous versions*. Then click *Create
rule.*

To enable CloudWatch logs for API Gateway, an IAM role must be created. This

> needs to be done once per region. Deployments with CloudWatch enabled  are

Reviewer 2: ```suggestion
needs to be done once per region. Deployments with CloudWatch enabled are
```

> automatically covered by this configuration.

Reviewer 2: Don't understand the last sentence.

1. Navigate to the *IAM Management Console*

2. Click *Roles* under *Access management*

3. Click *Create role*

4. Click *AWS service* for *Select type of trusted entity*

5. Click *API Gateway* for *Choose a use case*, click *Next: Permissions*

6. Ensure *AmazonAPIGatewayPushToCloudWatchLogs* is listed exclusively as the
policy and click *Next: Tags*

7. Add *name: azul-api_gateway* as a tag and click *Next: Review*

8. For *Role name* enter `azul-api_gateway`, ensure *Policies* are as prescribed
in step 6 and click *Create role*

> 9. Be sure to copy the *Role ARN* for the newly created `azul-api_gateway` role

Reviewer 2: ```suggestion
9. Copy the *Role ARN* for the newly created `azul-api_gateway` role
```

and go to the *API Gateway* console

> 10. Select any available Gateway resource (it doesn't matter which because this 

Reviewer 2: What if the current region has no gateways? Maybe there needs to be an intro paragraph stating that this must be done after at least one deployment was created in the configured AWS account and region.

Reviewer 2: ```suggestion
10. Select any available API Gateway resource (it doesn't matter which because this 
```

for *Number of days after objects become previous versions*. Then click *Create
rule.*

> To enable CloudWatch logs for API Gateway, an IAM role must be created. This
> can be done after at least one deployment is created in the configured AWS
> account and region, and is required once per region. With this configuration all
> deployments in API Gateway assume the created IAM role, which allows API Gateway
> to write logs in CloudWatch.

Reviewer 2: ```suggestion
To enable CloudWatch logs for API Gateway, an IAM role must be created and configured in the API Gateway console. This
must be done for each AWS account and region, after at least one Azul deployment has been created in that account and region. Once these steps have been completed, all
API Gateway instances in a region assume the resulting IAM role, allowing those instances
to write to CloudWatch Logs in that region.
```
